fastjet/internal/predicates/incircle.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package predicates
+
+import (
+	"math/big"
+)
+
+const (
+	// macheps is the machine epsilon aka unit roundoff
+	// it is the upper bound on relative error due to rounding
+	macheps = 2.22e-16
+)
+
+// Incircle determines whether the point (x,y) lies inside the circle formed
+// by the three points (x1,y1),(x2,y2) and (x3,y3)
+func Incircle(x1, y1, x2, y2, x3, y3, x, y float64) bool {
+	det, e := simpleIncircle(x1, y1, x2, y2, x3, y3, x, y)
+	if det > e {
+		return false
+	}
+	if det < -e {
+		return true
+	}
+	return robustIncircle(setBig(x1), setBig(y1), setBig(x2), setBig(y2), setBig(x3), setBig(y3), setBig(x), setBig(y))
+}
+
+// simpleIncircle finds the determinant of the matrix using the simple float64 type
+// |1 x1 y1 x1^2+y1^2|
+// |1 x2 y2 x2^2+y2^2|
+// |1 x3 y3 x3^2+y3^2|
+// |1 x  y  x^2 +y^2 |
+func simpleIncircle(x1, y1, x2, y2, x3, y3, x, y float64) (det, e float64) {
+	m := [][]float64{
```

Comment:
I'm going to go with recasting m, because it seems very complicated to calculate the max error for mat.Det